web/assets/js/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,70 @@
+class GoogleAnalyticsService {
+
+    constructor () {
+        this.gaDataAttr = 'data-ga-action';
+        this.gaCategoryAttr = 'data-ga-category';
+        this.gaLabelAttr = 'data-ga-label';
+    }
+
+    rewriteByChildSelector (selector) {
+        if (selector.classList.contains('ga-link-listener')) {
+            const childSelector = selector.querySelector('a');
+
+            return this.rewriteDataAttributes(selector, childSelector);
+        }
+
+        return selector;
+    }
+
+    getGAItemObject (selector) {
+        return {
+            hitType: 'event',
+            eventCategory: this.getGACategory(),
+            eventAction: this.getGAAction(selector),
+            eventLabel: this.getGALabel(selector)
+        };
+    }
+
+    getGAAction (selector) {
+        return selector.getAttribute(this.gaDataAttr);
+    }
+
+    getGACategory () {
+        return Utils.capitalize(document.body.getAttribute(this.gaCategoryAttr));
+    }
+
+    getGALabel (selector) {
+        return selector.getAttribute(this.gaLabelAttr);
+    }
+
+    sendGARequest (gaItem) {
+        ga(
+            'send',
+            gaItem.hitType,
+            gaItem.eventCategory,
+            gaItem.eventAction,
+            gaItem.eventLabel,
+            gaItem.transport
+        );
+    }
+
+    rewriteDataAttributes (parentSelector, childSelector) {
+        childSelector.setAttribute(this.gaDataAttr, this.getGAAction(parentSelector));
+        childSelector.setAttribute(this.gaLabelAttr, this.getGALabel(parentSelector));
+
+        return childSelector;
+    }
+
+    init () {
+        const gaSelectors = document.querySelectorAll(`[${this.gaDataAttr}]`);
+
+        gaSelectors.forEach(selector => {
+            selector = this.rewriteByChildSelector(selector);
```

Comment:
Just a nitpick: The name `selector` is confused for me. In this variable, I would expect a string with a selector like `.class`, but maybe it's just my impression.